### PR TITLE
Add MID to createTracksVariadic + fixes

### PR DESCRIPTION
**Pull request by shahor02**

### CODE REVIEW

File: DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainerCreateTracksVariadic.h [hunk 1]
```diff
@@ -326,25 +331,49 @@ void o2::globaltracking::RecoContainer::createTracksVariadic(T creator) const
...
   }
 
+  // MID standalone tracks
+  {
+    const auto& rofs = getMIDTracksROFRecords();
+    for (const auto& rof : rofs) {
+      auto bcWidth = 56;
+      auto rofMeanBC = rof.interactionRecord.differenceInBC(startIR) + bcWidth / 2;
+      float t0 = rofMeanBC * o2::constants::lhc::LHCBunchSpacingMUS;
+      float t0err = o2::constants::lhc::LHCBunchSpacingMUS * bcWidth / 2;
```
Comment (shahor02) [inline DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainerCreateTracksVariadic.h]: @dstocco I've borrowed t0 and t0err definition from the MCH, most probably wrong. Could you please check?

@jmyrcha for your ED tests small error in time should not be a problem.

Comment (dstocco) [reply DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainerCreateTracksVariadic.h]: Hi @shahor02 , differently from MCH the MID resolution allows to separate each BC. So we do not need an average. The BC in the ROF is the BC of the event.

Comment (shahor02) [reply DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainerCreateTracksVariadic.h]: Hi @dstocco 
Shall I take 0.6ns (bc length) ad an error estimate?

Comment (dstocco) [reply DataFormats/Detectors/GlobalTracking/include/DataFormatsGlobalTracking/RecoContainerCreateTracksVariadic.h]: Not sure why one would need an error on this. But yes, you can take the bc length as error for the moment.
I'll have a deeper look at the code tomorrow to understand it better.

**Pull request merged**